### DISCUSSION TITLE
How to get wallet ID of connected wallet to web3Modal?

### DISCUSSION THREAD

**Discussion by benreichman** [WalletConnect/walletconnect-monorepo#3300] [Developer Support] [answered]
I see that each wallet has a specific ID, as shown at this link https://walletconnect.com/explorer? . Is it possible to getthe id of a wallet that's been connected through the web3modal? For example, if a user uses rainbow wallet to connect through the QR code, is it possible to view that the user used rainbow to connect? Thanks!

**Comment by benreichman** [comment, marked as answer]
Solved. Leaving this for future reference in case anyone is trying to accomplish similar. Local storage -> 'wc@2:core:0.3//pairing' , object [0] contains 'peerMetadata' key which stores the name of the connected wallet.

**Reply by glitch-txs** [reply]
you can also get this from the `provider.session.peer`
